straxen/plugins/acqmon_processing.py

```diff
@@ -0,0 +1,131 @@
+import strax
+import straxen
+import numba
+import numpy as np
+from numba import njit
+from immutabledict import immutabledict
+
+export, __all__ = strax.exporter()
+
+#### Veto hardware ####:
+# V1495 busy veto module: 
+# Generates a 25 ns NIM pulse whenever a veto begins and a 25 ns NIM signal when it ends. 
+# A new start signal can occur only after the previous busy instance ended.
+# 1ms (1e6 ns) - minimum busy veto length, or until the board clears its memory
+    
+# DDC10 High Energy Veto:
+# 10ms (1e7 ns) - fixed HE veto length in XENON1T DDC10, 
+# in XENONnT it will be calibrated based length of large S2 SE tails
+# The start/stop signals for the HEV are generated by the V1495 board
+  
+
+@strax.takes_config(
+    strax.Option('max_veto_gap', default = int(5e8),
+                 help = 'Maximum separation between veto stop and start pulses [ns]'),
+    strax.Option('channel_map', track = False, type = immutabledict, 
+                 help = 'frozendict mapping subdetector to (min, max)'
+                      'channel number.'),
+    strax.Option('run_start_time', type = float, track = False, 
+                 default = int(0), help='time of start run (s since unix epoch)')
+)
+
+@export
+class VetoIntervals(strax.OverlapWindowPlugin):
+    """ Find pairs of veto start and veto stop signals and the veto duration between them
+    busy_*  <= V1495 busy veto for tpc channels
+    bb_*    <= V1495 busy veto for high energy tpc channels
+    hev_*   <= DDC10 hardware high energy veto
+    """
+        
+    __version__ ='0.0.19'
+    depends_on = ('raw_records_aqmon') 
+    provides  = ('veto_intervals')
+    data_kind = ('veto_intervals')
+
+
+    aqmon_channel_names = ('sum_wf','m_veto_sync',
+                           'hev_stop', 'hev_start', 
+                           'bb_stop','bb_start',
+                           'busy_stop', 'busy_start')
+    
+    tmp_dtype = strax.time_fields + [(('Interval between start and stop in [ns]', 'interval'), np.int64)]
+
+    def infer_dtype(self):
+        dtype = [(('faux time [ns]', 'time'), np.int64)]
+        dtype += [(('faux endtime [ns]', 'endtime'), np.int64)]
+        for veto in ('busy_','hev_','bb_'):
+            dtype += [(('Start '+ veto +'veto time since unix epoch [ns]', veto + 'time'), np.int64)]
+            dtype += [(('Stop '+ veto +'veto time since unix epoch [ns]', veto + 'endtime'), np.int64)]
+            dtype += [(('Duration of '+ veto +'veto time since unix epoch [ns]', veto + 'interval'), np.int64)]
+        return dtype
+    
+    def setup(self):
+        self.channel_range = self.config['channel_map']['aqmon']
+        self.channel_numbers = np.arange(self.channel_range[1]-7,self.channel_range[1] + 1, 1)
+        self.channel_map = dict(zip(self.aqmon_channel_names,self.channel_numbers))
+        self.t0 = self.config['run_start_time']
```

Comment:
This might be needed depending on what we want to do with inserting these data to mongo. If we save them here or in another plugin to a dedicated collection we need the run_id imho.

Reply:
Neither of these quantities are the run id, though.

Reply:
Sorry, got confused for a sec...doing smth else in parallel. I've explained the rest above, the run start time is a relic from the times I wanted to get dynamic deadtime calculation in this plugin, I can remove it.